app/src/androidTest/java/com/sdp/movemeet/view/activity/ActivityDescriptionActivityTest.java

```diff
@@ -90,31 +106,53 @@ public void onComplete(@NonNull Task<AuthResult> task) {
             assert (false);
         }
 
-        ActivityScenarioRule<ActivityDescriptionActivity> testRule = new ActivityScenarioRule<>(new Intent(getApplicationContext(), ActivityDescriptionActivity.class).putExtra("activity", activity));
-    }
+        Intent intent = new Intent(getApplicationContext(), ActivityDescriptionActivity.class).putExtra("activity", activity);
+
+
+        ActivityScenario testRule = ActivityScenario.launch(intent);
+
+        onView(withId(R.id.activity_title_description)).check(matches(withText(DUMMY_TITLE)));
+        //onView(withId(R.id.activity_date_description)).check(matches(withText(String.valueOf(DUMMY_DATE))));
+        onView(withId(R.id.activity_address_description)).check(matches(withText(DUMMY_ADDRESS)));
+        onView(withId(R.id.activity_sport_description)).check(matches(withText(String.valueOf(DUMMY_SPORT))));
+        //onView(withId(R.id.activity_duration_description)).check(matches(withText(String.valueOf(DUMMY_DURATION))));
+
+        onView(withId(R.id.activity_organisator_description)).check(matches(withText(DUMMY_ORGANISATOR_ID)));
+        //onView(withId(R.id.activity_number_description)).check(matches(withText(String.valueOf(DUMMY_NUMBER_PARTICIPANT))));
+        //onView(withId(R.id.activity_participants_description)).check(matches(withText(String.valueOf(DUMMY_PARTICIPANTS_ID.size()))));
+
+        try {
+            Thread.sleep(1500);
+        } catch (InterruptedException e) {
+            assert (false);
+        }
 
-    /*@Before
-    public void setUp(){
-        fStore = mock(FirebaseFirestore.class);
-        storageReference = mock(StorageReference.class);
-        activity = mock(Activity.class);
+    }
 
-    }*/
 
     @Test
     public void create() {
         Intents.init();
         Intents.release();
     }
 
-    /*@Test
-    public void createTitleViewTest(){
-        ViewGroup view = inflate(R.layout.activity_description);
-        TextView textView = view.findViewById(R.id.activity_title_description);
+    @Test
+    public void chatButtonIsCorrect() {
+        Intents.init();
```

Comment:
Couldn't the init for Intents be refactored in a @Before, since it is used in many tests ? And the release could be done in the @After

Reply:
It's true, I will do it in my next pull request